### PR TITLE
Fixes 111: record package count

### PR DESCRIPTION
**Pull request by jlsherrill**

### CODE REVIEW

**Comment by jlsherrill** [comment]
https://issues.redhat.com/browse/HMSCONTENT-111

**Comment by avisiedo** [comment]
At `pkg/dao/repositories.go` method `Update` needs the last line below:

```go
	repo.URL = repoIn.URL
	repo.Revision = repoIn.Revision
	repo.PackageCount = repoIn.PackageCount
```

**Review by rverdile — COMMENTED** [review]
DeepCopyInto may also need an update

**Comment by avisiedo** [comment]
I launched the below successfully and I got the package_count into the database:

```raw
make db-clean clean build db-up repos-import
./release/external-repos introspect-all
make db-cli-connect
select url, package_count from repositories;
```
branch need rebase; no further comments

**Comment by jlsherrill** [comment]
note, this is currently broken by https://issues.redhat.com/browse/HMSCONTENT-185

**Comment by jlsherrill** [comment]
need to update this to use a migration!

**Comment by jlsherrill** [comment]
This should be good for re-review @avisiedo

File: db/migrations/20220923124812_add_package_count.down.sql [hunk 1]
```diff
@@ -0,0 +1,5 @@
+BEGIN;
+
+alter table repositories drop column package_count;
+
+COMMIT;
```
Comment (avisiedo) [inline db/migrations/20220923124812_add_package_count.down.sql:5]: The new column was added / removed with the migrations; tried by:

```sh
make db-clean clean build db-up
./release/dbmigrate down -step 1
make db-cli-connect
\d repositories
\q
./release/dbmigrate up -step 1
make db-cli-connect
\d repositories
\q
```
working as expected :+1:

File: pkg/dao/repositories_test.go [hunk 2]
```diff
@@ -106,6 +106,7 @@ func (s *RepositorySuite) TestUpdateRepository() {
 		LastIntrospectionUpdateTime:  &expectedTimestamp,
 		LastIntrospectionError:       pointy.String("expected error"),
 		Status:                       config.StatusUnavailable,
+		PackageCount:                 123,
```
Comment (avisiedo) [inline pkg/dao/repositories_test.go:109]: I would update the `repoPublicTest` and `repoPrivateTest` template variables at `pkg/dao/suite_test.go` to add `PackageCount` field (using values such as `987` and `543`), that should arise a few points to update tests at:

- TestFetchForUrl (Here I have realized a previous confusion usage of `s.repo` and `s.repoPrivate` not related with this change; should I add a separate ticket? it is when comparing the `s.repoPrivate` fetched which is using the `s.repo` instead).
- TestList
- TestUpdateRepository

**Review by avisiedo — CHANGES_REQUESTED** [review]
Only a few changes at unit tests, and a question if creating a new ticket for some previous change that I have realized reviewing this pr.

the rest lgtm :+1:

File: pkg/dao/repositories_test.go (hunk 2 above)
Comment (jlsherrill) [reply pkg/dao/repositories_test.go:113]: updated, and went ahead and fixed that mixed up confusion on the repo/privateRepo in TestFetchForUrl

**Comment by avisiedo** [comment]
lgtm

**Review by avisiedo — APPROVED** [review]

**Review by avisiedo — COMMENTED** [review]
lgtm

I only see the bot is raising some errors

```raw
ApiRepositoryResponse has no attribute 'package_count' at ['received_data']['package_count']
```

I didn't realize about the qe-testing-needed label.

**Review by avisiedo — APPROVED** [review]
lgtm

**Review by Andrewgdewar — APPROVED** [review]
Screen Shot 2022-09-27 at 2 57 31 PM

**Comment by mshriver** [comment]
Merging before heavy QE given priorities and tasking.

**Pull request merged**